AIO/src/org/aio/activities/quests/SheepShearer.java

```diff
@@ -0,0 +1,150 @@
+            "Of course!",
+            "I'm something of an expert actually!"
+    };
+
+    private static final int INVENTORY_SLOTS_REQUIRED = 22;
+
+    private static final String[] ITEMS_NEEDED = {
+            "Ball of wool",
+            "Wool",
+            "Shears"
+    };
+
+    private final DepositAllBanking depositAllBanking = new DepositAllBanking(ITEMS_NEEDED);
+
+    public SheepShearer() {
+        super(Quest.SHHEP_SHEARER);
+    }
+
+    @Override
+    public void onStart() {
+        depositAllBanking.exchangeContext(getBot());
+
+        makeAllInterface = new MakeAllInterface("Ball of Wool");
+        makeAllInterface.exchangeContext(getBot());
+    }
+
+    @Override
+    public void runActivity() throws InterruptedException {
+        if (getInventory().getEmptySlotCount() < INVENTORY_SLOTS_REQUIRED - getInventory().getAmount(ITEMS_NEEDED)) {
+            depositAllBanking.run();
+        } else if (getTabs().getOpen() != Tab.INVENTORY) {
+            getTabs().open(Tab.INVENTORY);
+        } else {
+            switch (getProgress()) {
+                case 0:
+                    talkToFarmer();
+                    break;
+                case 1:
+                    if (hasRequiredItems()) {
+                        talkToFarmer();
+                    } else {
+                        getItemsNeeded();
+                    }
+                    break;
+                case 21:
+                    log("Quest is complete");
+                    isComplete = true;
+                    break;
+                default:
+                    log("Unknown progress config value: " + getProgress());
+                    setFailed();
+                    break;
+            }
+        }
+    }
+
+    private boolean hasRequiredItems() {
+        return getInventory().getAmount("Ball of wool") > 19;
+    }
+
+    private void getItemsNeeded() throws InterruptedException {
+        if (!getInventory().contains("Shears")) {
+            getGroundItem(FARMER, "Shears");
+        } else if (!getInventory().contains("Ball of wool") && getInventory().getAmount("Wool") < 20) {
+            getItemFromNPC(SHEEP, "Shear");
+        } else if (getInventory().getAmount("Ball of wool") < 20) {
+            useObject(SPINNER, "Spin");
+        }
+    }
+
+    private void getItemFromNPC(Area place, String interaction) throws InterruptedException {
+        if (place.contains(myPlayer())) {
+            NPC npc = getNpcs().closest(n -> !n.hasAction("Talk-to") && n.hasAction(interaction) && place.contains(n));
+            if (npc != null && npc.interact(interaction)) {
+                Sleep.sleepUntil(() -> myPlayer().getAnimation() == 893, 15000);
+                Sleep.sleepUntil(() -> !myPlayer().isAnimating(), 10000);
+            }
+        } else {
+            getWalking().webWalk(place);
+        }
+    }
+
+    private void useObject(Area place, String objectAction) throws InterruptedException {
```

Comment:
nit: This function is named `useObject` but it really means `spinWool`

Maybe rename this function, or make it more akin to what it's actually doing?

Reply:
Will take it out to a more generic function when I refactor and make the helper class.